### PR TITLE
7839: Remove contactPrimary and instead add it to an array of petitioners on a Case

### PR DESCRIPTION
**Pull request by rachelschneiderman**

### CODE REVIEW

File: shared/src/business/entities/DocketEntry.js [hunk 1]
```diff
@@ -207,9 +207,9 @@ DocketEntry.prototype.setAsServed = function (servedParties = null) {
  * generates the filedBy string from parties selected for the document
```
Comment (mrinsin) [inline shared/src/business/entities/DocketEntry.js:207]: Update JSdocs

File: shared/src/business/entities/cases/Case.js [hunk 2]
```diff
@@ -1413,6 +1420,31 @@ const isAssociatedUser = function ({ caseRaw, user }) {
   );
 };
 
+// 7839 TODO - docs
```
Comment (mrinsin) [inline shared/src/business/entities/cases/Case.js]: JSdocs

File: shared/src/business/entities/cases/Case.js [hunk 3]
```diff
@@ -1413,6 +1420,31 @@ const isAssociatedUser = function ({ caseRaw, user }) {
...
+Case.prototype.getContactPrimary = function () {
+  return getContactPrimary(this);
+};
+
+const updatePetitioner = function (rawCase, updatedPetitioner) {
+  const foundPetitioner = rawCase.petitioners.find(
+    p => p.contactId === updatedPetitioner.contactId,
+  );
+
+  if (foundPetitioner) {
```
Comment (mrinsin) [inline shared/src/business/entities/cases/Case.js:1453]: Throw an error if no petitioner found, handled by interactor

File: shared/src/business/entities/cases/CaseExternalForInternationalContacts.test.js [hunk 4]
```diff
@@ -43,31 +46,34 @@ describe('CaseExternal', () => {
         { applicationContext },
       );
       expect(caseExternal.getFormattedValidationErrors()).toEqual({
-        contactPrimary: { country: contactErrorMessages.country },
+        petitioners: [{ country: contactErrorMessages.country, index: 0 }],
       });
     });
 
-    it('can validate primary contact', () => {
+    it('can validate the primary contact in the petitioners arrray', () => {
```
Comment (mrinsin) [inline shared/src/business/entities/cases/CaseExternalForInternationalContacts.test.js]: Nit: typo

File: shared/src/business/test/createCaseFromPaperInteractor.e2e.test.js [hunk 5]
```diff
@@ -119,6 +116,13 @@ describe('createCaseFromPaperInteractor integration test', () => {
       orderForOds: false,
       orderForRatification: false,
       orderToShowCause: false,
+      petitioners: [
```
Comment (mrinsin) [inline shared/src/business/test/createCaseFromPaperInteractor.e2e.test.js:120]: Retrieve contactPrimary and then add assertions (to equal newContactId, to NOT equal oldContactId

File: shared/src/business/useCases/getCaseInteractor.test.js [hunk 6]
```diff
@@ -197,8 +204,8 @@ describe('getCaseInteractor', () => {
       const result = await getCaseInteractor(applicationContext, {
         docketNumber: '101-18',
       });
-      expect(result.contactPrimary.city).toBeDefined();
-      expect(result.contactPrimary.sealedAndUnavailable).toBe(false);
+      expect(getContactPrimary(result).city).toBeDefined();
```
Comment (mrinsin) [inline shared/src/business/useCases/getCaseInteractor.test.js]: Save contactPrimary in a const and expect on that const

File: shared/src/business/useCases/public/getPublicCaseInteractor.test.js [hunk 7]
```diff
@@ -93,16 +94,19 @@ describe('getPublicCaseInteractor', () => {
...
     ).toHaveBeenCalled();
     expect(result).toMatchObject({
-      contactPrimary: {
-        name: MOCK_CASE.contactPrimary.name,
-        state: MOCK_CASE.contactPrimary.state,
-      },
       docketNumber: '123-45',
+      petitioners: [
+        {
+          name: getContactPrimary(MOCK_CASE).name,
```
Comment (mrinsin) [inline shared/src/business/useCases/public/getPublicCaseInteractor.test.js]: Save contactPrimary to a const and access properties off it

File: web-client/src/presenter/actions/getUserPendingEmailAction.js [hunk 8]
```diff
@@ -11,7 +11,9 @@ export const getUserPendingEmailAction = async ({
   applicationContext,
   get,
 }) => {
-  const contactId = get(state.caseDetail.contactPrimary.contactId);
+  const { contactId } = applicationContext
```
Comment (mrinsin) [inline web-client/src/presenter/actions/getUserPendingEmailAction.js]: Add null check for contactId

File: web-client/src/presenter/actions/setupContactPrimaryFormAction.js [hunk 9]
```diff
@@ -7,8 +7,16 @@ import { state } from 'cerebral';
  * @param {object} providers.props the cerebral props object
  * @param {object} providers.store the cerebral store
  */
-export const setupContactPrimaryFormAction = ({ props, store }) => {
-  store.set(state.form.docketNumber, props.caseDetail.docketNumber);
-  store.set(state.form.contactPrimary, props.caseDetail.contactPrimary);
-  store.set(state.form.partyType, props.caseDetail.partyType);
+export const setupContactPrimaryFormAction = ({
```
Comment (mrinsin) [inline web-client/src/presenter/actions/setupContactPrimaryFormAction.js:11]: Jsdocs

**Review by mrinsin — APPROVED** [review]

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**